packages/hadron-document/src/element.ts

```diff
@@ -766,61 +702,34 @@ export class Element extends EventEmitter {
   /**
    * Is the element empty?
    *
-   * @param {Element} element - The element to check.
+   * @param element - The element to check.
    *
-   * @returns {Boolean} If the element is empty.
+   * @returns If the element is empty.
    */
-  _isElementEmpty(this: unknown, element: Element | undefined | null): boolean {
+  _isElementEmpty(element: Element | undefined | null): boolean {
     return !!element && element.isAdded() && element.isBlank();
   }
 
   /**
    * Check if the value is expandable.
    *
-   * @param {Object} value - The value to check.
+   * @param value - The value to check.
    *
-   * @returns {Boolean} If the value is expandable.
+   * @returns If the value is expandable.
    */
-  _isExpandable(
-    this: unknown,
-    value: BSONValue
-  ): value is BSONObject | BSONArray {
+  _isExpandable(value: BSONValue): value is BSONObject | BSONArray {
     return isObject(value) || isArray(value);
   }
 
   /**
    * Generates a sequence of child elements.
    *
-   * @param {Object} object - The object to generate from.
-   *
-   * @returns {Array} The elements.
-   */
-  _generateElements(object: BSONObject | BSONArray): LinkedList {
-    return new LinkedList(this, object); // eslint-disable-line no-use-before-define
-  }
-
-  /**
-   * Get the key for the element.
+   * @param object - The object to generate from.
    *
-   * @param {String} key
-   * @param {Number} index
-   *
-   * @returns {String|Number} The index if the type is an array, or the key.
-   */
-  _key(key: string, index: number): string | number {
-    return this.currentType === 'Array' ? index : key;
-  }
-
-  /**
-   * Add a new element to the parent.
+   * @returns The elements.
    */
-  _next(): void {
-    if (
-      !this._isElementEmpty(this.nextElement) &&
-      !this._isElementEmpty(this)
-    ) {
-      this.parent!.insertAfter(this, '', '');
-    }
+  _generateElements(object: BSONObject | BSONArray): ElementList {
+    return new ElementList(this, object); // eslint-disable-line no-use-before-define
```

Comment:
I feel like this lint rule is pretty silly in situations like this. Looking at the docs, I don’t see an obvious way to tune this so it doesn’t have false positives like this, but maybe we could just turn it off for classes and functions altogether?

Reply:
Makes sense, let's disable it for classes and fns 👍

Reply:
Ah, actually I think it's already not applied (at least not in ts files), maybe this was a leftover from converting the code

Reply:
Cleaned-up the eslint comment, we should probably do the same thing that mongosh does and run eslint with this flag that checks for those